src/openzaak/components/zaken/api/validators.py

```diff
@@ -280,6 +281,94 @@ def __call__(self, attrs: dict):
                 raise serializers.ValidationError(self.message, code=self.code)
 
 
+class DeelzaakReopenValidator:
+    """
+    Validate that the hoofdzaak is not closed  when a deelzaak is reopened.
+    """
+
+    code = "hoofdzaak-closed"
+    message = (
+        "De hoofdzaak is afgesloten."
+        "Een deelzaak kan pas worden geopend als de hoofzaak dat al is."
+    )
+
+    def __call__(self, attrs: dict):
+        if attrs.get("__is_eindstatus"):
+            return
+
+        zaak = attrs.get("zaak")
+        if not zaak or not zaak.hoofdzaak or not zaak.hoofdzaak.current_status:
+            return
+
+        if zaak.hoofdzaak.current_status.statustype.is_eindstatus():
```

Comment:
This should maybe only happen if the afleidingswijze == 'hoofdzaak'?

Reply:
I think this validation should apply regardless of afleidingswijze, because having an open deelzaak implies that the hoofdzaak is also not yet done (it's the inverse of the validation that validates all deelzaken are closed when you try to close the hoofdzaak)